src/libraries/System.Private.CoreLib/src/System/Environment.GetFolderPathCore.Unix.cs

```diff
@@ -37,20 +37,21 @@ private static string GetFolderPathCore(SpecialFolder folder, SpecialFolderOptio
             {
                 return string.Empty;
             }
-            else
-            {
-                Debug.Assert(option == SpecialFolderOption.Create);
 
-                Func<string, object> createDirectory = LazyInitializer.EnsureInitialized(ref s_directoryCreateDirectory, static () =>
-                {
-                    Type dirType = Type.GetType("System.IO.Directory, System.IO.FileSystem, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a", throwOnError: true)!;
-                    MethodInfo mi = dirType.GetTypeInfo().GetDeclaredMethod("CreateDirectory")!;
-                    return mi.CreateDelegate<Func<string, object>>();
-                });
-                createDirectory(path);
+            Debug.Assert(option == SpecialFolderOption.Create);
 
-                return path;
+            Func<string, object>? createDirectory = Volatile.Read(ref s_directoryCreateDirectory);
+            if (createDirectory is null)
+            {
+                Type dirType = Type.GetType("System.IO.Directory, System.IO.FileSystem, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a", throwOnError: true)!;
+                MethodInfo mi = dirType.GetTypeInfo().GetDeclaredMethod("CreateDirectory")!;
```

Comment:
A bit unrelated but this will mark/keep all methods on System.IO.Directory. You could remove GetTypeInfo() to avoid marking also private stuff but perhaps the type should be moved to SPC.

Reply:
Thanks. Fixed in https://github.com/dotnet/runtime/pull/44414.